src/query/plan/rule_based_planner.hpp

```diff
@@ -321,11 +321,18 @@ class RuleBasedPlanner {
 
   storage::EdgeTypeId GetEdgeType(EdgeTypeIx edge_type) { return context_->db->NameToEdgeType(edge_type.name); }
 
-  std::vector<storage::EdgeTypeId> GetEdgeTypes(std::vector<EdgeTypeIx> edge_types) {
+  std::vector<storage::EdgeTypeId> GetEdgeTypes(std::vector<QueryEdgeType> edge_types) {
     std::vector<storage::EdgeTypeId> transformed_edge_types;
     transformed_edge_types.reserve(edge_types.size());
+
     for (const auto &type : edge_types) {
-      transformed_edge_types.push_back(GetEdgeType(type));
+      if (const auto *edge_type_atom = std::get_if<EdgeTypeIx>(&type)) {
+        transformed_edge_types.push_back(GetEdgeType(*edge_type_atom));
+      } else {
+        throw QueryException(
+            "Failed to work with dynamic edge types! Please contact Memgraph support as this scenario should not "
+            "happen!");
```

Comment:
`"Failed to work with dynamic edge types! Avoid using parameters or variables when creating a new edge. Also, please contact Memgraph support, as this scenario should not happen (this is an unexpected code path, but we didn't want to just crash Memgraph, help us improve the query planner)!"`

@Ana-Marija-design this is an interesting error message to craft 😃